extensions/wikia/CuratedContent/maintenance/SpecialPromoteToCuratedContent.php

```diff
@@ -0,0 +1,163 @@
+
+class SpecialPromoteToCuratedContentMigrator extends Maintenance {
+	public function __construct() {
+		$this->addOption( 'dry-run', 'Dry run - do not save changes to curated content' );
+		parent::__construct();
+	}
+
+	public function execute() {
+		global $wgCityId, $wgLang, $wgDBname;
+
+		$dryRun = $this->hasOption('dry-run');
+
+		$change = false;
+		$cv = new CityVisualization();
+		$communityDataService = new CommunityDataService( $wgCityId );
+
+		$curatedContentData = [
+			'curated' => $communityDataService->getCurated(),
+			'featured' => $communityDataService->getFeatured(),
+			'optional' => $communityDataService->getOptional(),
+			'community_data' => $communityDataService->getCommunityData()
+		];
+
+		if ( empty( $curatedContentData['community_data']['description'] ) ) {
+			$originalDescription = $curatedContentData['community_data']['description'];
+			// updateDescription
+			$description = $this->getPromoteDescription( $cv, $wgCityId, $wgLang->getCode() );
+		} else {
+			$originalDescription = '';
+		}
+
+		if ( empty( $curatedContentData['community_data']['image_id'] ) ) {
+			$originalImageId = $curatedContentData['community_data']['image_id'];
+
+			// update image
+			$image = ( new PromoImage( PromoImage::MAIN, $wgDBname ) );
+			$corporateWikiId = $cv->getTargetWikiId( $wgLang->getCode() );
+			$imageTitle = $image->getOriginFile()->getTitle();
+			$imageId = $imageTitle->exists() ? $imageTitle->getArticleID() : null;
+
+			if ( !empty( $imageId ) ) {
+				// we are good to go; image exists on the destination wiki
+			} else {
+				// image does not exist on this community
+				// we need to upload the image to the local wiki
+				// we first need to fetch it from the corporate wiki
+				$imageUrl = $image->getOriginFile( $corporateWikiId )->getUrl();
+				$imageName = $image->getPathname();
+
+				$this->disableUploadValidationHook();
+				$result = $this->uploadImageToLocalWiki( $imageName, $imageUrl );
+
+				if ( !empty( $result['page_id'] ) ) {
+					$imageId = $result['page_id'];
+				} else {
+					$this->output( 'Upload failed: ' . serialize( $result ) );
+				}
+			}
+
+		}
+
+		if(!empty($description)) {
+			$curatedContentData['community_data']['description'] = $description;
+			$this->output( sprintf(
+					"\nPreparing to update description from:\n%s\nto\n%s\n",
+					empty( $originalDescription ) ? "''" : $originalDescription,
+					$description)
+			);
+			$change = true;
+		}
+
+		if(!empty($imageId)) {
+			$curatedContentData['community_data']['image_id'] = $imageId;
+			$this->output( sprintf(
+					"\nPreparing to update image from %d to %d\n",
+					$originalImageId,
+					$imageId)
+			);
+			$change = true;
+		}
+
+		if ( $dryRun ) {
+			$this->output( "\nDRY RUN: Skipping save of Curated Content\n" );
+		} else {
+			if ( $change ) {
+				$this->output( "\nSaving Curated Content\n" );
+				$result = $communityDataService->setCuratedContent( $curatedContentData );
```

Comment:
Yes :)
As in https://github.com/Wikia/app/pull/9623/files#diff-49cc6a229edc8757921429da24e5a25eR27 - I can either rebase my brach onto yours, or modify per those two comments once DAT-3638b is merged